data/campaigns/Descent_Into_Darkness/scenarios/11_Descent_into_Darkness.cfg

```diff
@@ -391,419 +521,3071 @@
+            [command]
+                {NOTRAIT_UNIT 3 "Giant Scorpling" $loc.x $loc.y} {GUARDIAN}
+            [/command]
+        [/random_placement]
+
+        [random_placement]
+            num_items=1
+            variable=loc
+            min_distance=11
+            [filter_location]
+                x,y=1-29,1-24
+                terrain=Uu,Uu^*
+                [not]
+                    [filter][/filter]
+                [/not]
+            [/filter_location]
+            [command]
+                {NOTRAIT_UNIT 3 "Giant Spider" $loc.x $loc.y} {GUARDIAN}
+            [/command]
+        [/random_placement]
+
+        [unit]
+            type=Dread Bat
+            side=3
+            x,y=2,23
+            max_hitpoints=60
+            hitpoints=60
+            max_moves=5
+            ai_special=guardian
+            moves=5
+            level=3
+        [/unit]
+
+        {MODIFY_UNIT (type=Giant Spider) max_moves 3}
+
+        # Previously recruited and new adepts can now also become liches
+        [lua]
+            code = <<
+                for i, u in ipairs(wesnoth.get_recall_units { type = 'Dark Adept DiD, Dark Sorcerer DiD' }) do
+                    if u.type == 'Dark Adept DiD' then
+                        u:transform('Dark Adept')
+                    elseif u.type == 'Dark Sorcerer DiD' then
+                        u:transform('Dark Sorcerer')
+                    end
+                end
+            >>
+        [/lua]
+
+        [disallow_recruit]
+            side=1
+            type=Dark Adept DiD
+        [/disallow_recruit]
+
+        [allow_recruit]
+            side=1
+            type=Dark Adept
+        [/allow_recruit]
+
+        {VARIABLE scenario_stage 2}
+        {VARIABLE puzzle_coffin 0}
+        {VARIABLE puzzle_potion 0}
+
+        [objectives]
+            side=1
+            [objective]
+                condition=win
+                description= _ "Regain your strength"
+            [/objective]
+            [objective]
+                condition=lose
+                description= _ "Destruction of Mal Keshar"
+            [/objective]
+
+            [gold_carryover]
+                bonus=no
+                carryover_percentage=100
+            [/gold_carryover]
+        [/objectives]
+    [/event]
+
+# text at lair entrance
+    [event]
+        name=moveto
+        first_time_only=no
+        [filter]
+            id=Mal Keshar
+            x,y=13,27
+        [/filter]
+        [filter_condition]
+            [variable]
+                name=scenario_stage
+                numerical_equals=2
+            [/variable]
+        [/filter_condition]
+
+        [message]
+            speaker=narrator
+            # wmllint: local spelling Zephryn
+            message= _ "<s><i>Welcome to Zephryn’s laboratory. Please be mindful of ongoing experiments during your visit.</i></s>
+<big><i><b>This territory is the domain of the mighty Xanthric. All unwelcome intruders shall die.</b></i></big>"
```

Comment:
By definition intruders are unwelcome... but I take it this tautology is deliberate as a sort of joke?